### ISSUE TITLE
DB ticket information not available after timetable change (Fahrplanwechsel)

### ISSUE THREAD

**Issue by grssnbchr** [public-transport/hafas-client#331]
While the hafas-client with DB profile is still able to find journeys after 2024-12-15, it always returns empty fare information after this date. 

Running this example with a journey that departs before 2024-12-15 still also returns ticket information,

https://github.com/public-transport/hafas-client/blob/e9211e8105049b39c00c62fb33c55990ec3e45e7/p/db/example.js

after this date, there's no ticketing information.

Any clue why this might be the case? Is a new auth / salt / version needed?

**Comment by derhuerst** [comment]
This is the `#opentransport:matrix.org` discussion about this: https://matrix.to/#/!fGsJOaXhgnTCFCghEl:matrix.org/$Af-pOVmexMHJ9b5YYtV2ThMl5loSI0oY9YaPhdP3GNU?via=matrix.org&via=kde.org&via=tchncs.de.

**Comment by traines-source** [comment]
TL;DR: It is not yet known what the exact reason for this is, but the legacy DB HAFAS API simply doesn't seem to return the prices anymore. Furthermore, it has been increasingly unstable in the past couple of weeks (lots of `CGI: server could not be reached` and similar errors). Since the new bahn.de and DB Navigator now use a different API, this might all hint at the legacy HAFAS API being turned off soon, although people say that there are no concrete plans for that inside DB.
I have started building a drop-in-replacement for hafas-client that relies on the new APIs of bahn.de amongst others, https://github.com/public-transport/db-vendo-client, but it is still very alpha and it remains to be seen how well it will work with regards to quotas etc (but it returns prices!).

**Comment by derhuerst** [comment]
cc @PaulSut

**Comment by grssnbchr** [comment]
> 
> It _seems_ to be that `refreshJourneys()` still has a working prices & tickets. Can you confirm that?
> 
> If that is the case, then we could automatically query it in `journeys()` if `opt.tickets` is `true`.
> 

I cannot fully follow this. How is refreshJourneys() different than journeys()? I tried to find this in the code base, but in the end, this also "just" calls the endpoint with `"meth": "Reconstruction"`, no?

**Comment by traines-source** [comment]
Can't reproduce that: `/journeys/T$A=1@O=Frankfurt(Main)Hbf@L=8000105@a=128@$A=1@O=Stuttgart%20Hbf@L=8000096@a=128@$202412181305$202412181438$ICE%20%20575$$1$$$$$$§W$A=1@O=Stuttgart%20Hbf@L=8000096@a=128@$A=1@O=Hauptbf%20(Arnulf-Klett-Platz),%20Stuttgart@L=561603@a=128@$202412181438$202412181450$$$1$$$$$$§T$A=1@O=Hauptbf%20(Arnulf-Klett-Platz),%20Stuttgart@L=561603@a=128@$A=1@O=Hölderlinplatz,%20Stuttgart@L=561469@a=128@$202412181451$202412181458$Bus%20%20%2040$$1$$$$$$`
returns no price. Maybe you checked before the timetable change?

And yes, hafas-client should stay a client for HAFAS, but my goal was for any downstream applications to be able to seamlessly replace hafas-client with db-vendo-client. And I figured it couldn't harm to keep the git history.

**Comment by derhuerst** [comment]
> Maybe you checked before the timetable change?

Oops, I did.

**Comment by derhuerst** [comment]
FYI: I have just published `hafas-client@6.3.3`, which updates the DB profile to `ver` to `1.78` and `ext` to `DB.R22.04.a`, but AFAICT this hasn't changed anything about the missing prices/fares/tickets data.

**Comment by derhuerst** [comment]
might be related, from https://github.com/schildbach/public-transport-enabler/pull/614#issuecomment-2577642791: 

> I think it has happened, DB HAFAS `mgate.exe` has been shut off for good, at least it now returns `503`. […]
